Sources/Segment/ObjC/ObjCEvents.swift

```diff
@@ -0,0 +1,372 @@
+//
+//  ObjCEvents.swift
+//  
+//
+//  Created by Brandon Sneed on 4/17/23.
+//
+
+#if !os(Linux)
+
+import Foundation
+
+internal protocol ObjCEvent {
+    associatedtype EventType
+    var _event: EventType { get set }
+}
+
+@objc(SEGDestinationMetadata)
+public class ObjCDestinationMetadata: NSObject {
+    internal var _metadata: DestinationMetadata
+    
+    public var bundled: [String] {
+        get { return _metadata.bundled }
+        set(v) { _metadata.bundled = v }
+    }
+    
+    public var unbundled: [String] {
+        get { return _metadata.unbundled }
+        set(v) { _metadata.unbundled = v }
+    }
+    
+    public var bundledIds: [String] {
+        get { return _metadata.bundledIds }
+        set(v) { _metadata.bundledIds = v }
+    }
+    
+    internal init?(_metadata: DestinationMetadata?) {
+        guard let m = _metadata else { return nil }
+        self._metadata = m
+    }
+    
+    init(bundled: [String], unbundled: [String], bundledIds: [String]) {
+        _metadata = DestinationMetadata(bundled: bundled, unbundled: unbundled, bundledIds: bundledIds)
+    }
+}
+
+@objc(SEGRawEvent)
+public protocol ObjCRawEvent: NSObjectProtocol {
+    var type: String? { get }
+    var messageId: String? { get }
+    var timestamp: String? { get }
+    var anonymousId: String? { get set }
+    var userId: String? { get set }
+    
+    var context: [String: Any]? { get set }
+    var integrations: [String: Any]? { get set }
+
+    var metadata: ObjCDestinationMetadata? { get set }
+}
+
+internal func eventFromObjCEvent(_ event: ObjCRawEvent?) -> RawEvent? {
+    guard let event = event as? (any ObjCEvent) else { return nil }
+    return event._event as? RawEvent
+}
+
+internal func objcEventFromEvent<T: RawEvent>(_ event: T?) -> ObjCRawEvent? {
+    guard let event = event else { return nil }
+    switch event {
+    case let e as TrackEvent:
+        return ObjCTrackEvent(event: e)
+    case let e as IdentifyEvent:
+        return ObjCIdentifyEvent(event: e)
+    case let e as ScreenEvent:
+        return ObjCScreenEvent(event: e)
+    case let e as GroupEvent:
+        return ObjCGroupEvent(event: e)
+    case let e as AliasEvent:
+        return ObjCAliasEvent(event: e)
```

Comment:
Do you have a page event that could go here?

Reply:
Not yet.